htdocs/core/tpl/extrafields_list_search_sql.tpl.php

```diff
@@ -59,13 +59,14 @@
 			if (in_array($typ, array('sellist')) && !is_numeric($crit)) {
 				$mode_search = 0;// Search on a foreign key string
 			}
-			if (in_array($typ, array('chkbxlst', 'checkbox'))) {
+			if (in_array($typ, array('chkbxlst', 'checkbox', 'select'))) {
 				$mode_search = 4; // Search on a multiselect field with sql type = text
 			}
 			if (is_array($crit)) {
 				$crit = implode(' ', $crit); // natural_search() expects a string
-			} elseif ($typ === 'select' and is_string($crit) and strpos($crit, ' ') === false) {
-				$sql .= " AND (".$extrafieldsobjectprefix.$tmpkey." = '".$db->escape($crit)."')";
+			} elseif ($typ === 'select' and is_string($crit) and strpos($crit, ',') === false) {
+				$critSelect = implode("','", array_map(array($db, 'escape'), explode(',', $crit)));
+				$sql .= " AND (".$extrafieldsobjectprefix.$tmpkey." IN ('".$critSelect."') )";
```

Comment:
When you have a sql forging a IN, the var $critSelect must be escaped with 
`$db->sanitize($critSelect, 1)`

Reply:
I already do a 
```array_map(array($db, 'escape'), explode(',', $crit))```

Reply:
and make a $db->sanitize after could sanitize an escape char so $db->sanitize could maybe unsecure query in this case, no ?